### ISSUE TITLE
Please upgrade "github.com/hashicorp/raft" to v1.1.2

### ISSUE THREAD

**Issue by onlyjob** [hashicorp/consul#7087]
Currently Consul 1.5.2 FTBFS with [github.com/hashicorp/raft](https://github.com/hashicorp/raft/tags) v1.1.2 as follows:

```
# github.com/hashicorp/consul/agent/consul 
src/github.com/hashicorp/consul/agent/consul/server.go:584:3: cannot use s.logger (type *log.Logger) as type hclog.Logger in field value: 
        *log.Logger does not implement hclog.Logger (missing Debug method) 
```

**Comment by s-christoff** [comment]
Howdy @onlyjob, thank you for bringing this to our attention.
In the more recent versions of Raft we were able to update the Raft logger to use [hclog](https://github.com/hashicorp/go-hclog).  Because of this we made this [update](https://github.com/hashicorp/consul/pull/7079) in Consul so it could work with Raft's hclogs. Please update Consul and it should be able to work with the updated Raft version.

**Comment by i0rek** [comment]
Closing since there is nothing left for us to do.

**Comment by onlyjob** [comment]
Thank you. Cherry-picking 5f57b95480f7667ed58392bf021029b3f8a0cb96 fixed the FTBFS with raft v1.1.2 - very helpful.

As for upgrading Consul, I'd very much like to do that but lack of harmony in Hashicorp ecosystem holds the progress back: https://github.com/hashicorp/nomad/issues/6785. Because of this issue with Nomad I can not upgrade Consul 1.5.2 in Debian...
